app/models/article.rb

```diff
@@ -552,6 +554,22 @@ def validate_collection_permission
     errors.add(:collection_id, "must be one you have permission to post to")
   end
 
+  def validate_co_authors
+    return unless co_author_ids
+    return unless co_author_ids.include?(user_id)
+
+    errors.add(:co_author_ids, "must not be the same user as the author")
+  end
+
+  def validate_co_authors_must_not_be_the_same
+    return unless co_author_ids
+
+    co_author = co_author_ids.map { |id| id }
+    return if co_author_ids.include?(co_author)
```

Comment:
> Added validations around co_author_ids to ensure that co_author_ids are not the same as the author ID and that co_author_ids cannot be duplicated (e.g. the same co-author cannot be added as a co-author to a particular article twice).

If I'm following along correctly I think this may be the problem. Both `co_author_ids` and `co_author` are arrays and the `include?` method won't work for this. Not sure if it's the best way but I think this suggestion may help validate there are no duplicate IDs in the array

```suggestion
return if co_author_ids.uniq.count == co_author_ids.count
```

The error message here should also maybe read something along the lines of "coauthor ids must be unique"

Reply:
🤦‍♀️ you are a lifesaver, @fdoxyz! I have been soooo stumped by why this was failing and any time I dropped into `pry` I found no useful information. I've made both of these great suggestions. Thank you times a million! 😄

Reply:
Happens all the time with these small things! Glad it helped 😄